Script/AtomicNET/AtomicNET/Core/Profiler.cs

```diff
@@ -12,12 +13,16 @@ public partial class Profiler : AObject
         {
 #if ATOMIC_PROFILING
             var profiler = AtomicNET.GetSubsystem<Profiler>();
-            profiler?.BeginBlock(name, file, line, color, (byte)status);
+            if (profiler != null)
```

Comment:
This caching of subsystem pointers seem to comes up rather often. Maybe it would be better idea for next PR to cache engine subsystems as weak pointers in `Context`?